src/transport/AdminPairingTable.cpp

```diff
@@ -258,15 +307,24 @@ CHIP_ERROR AdminPairingInfo::SetOperationalCert(const ByteSpan & cert)
         ReleaseOperationalCert();
     }
 
+    ByteSpan noc;
+    ByteSpan icac;
+    ConvertChipCertArrayToChipCerts(cert, noc, icac);
+
+    if (icac.data() != nullptr && icac.size() != 0)
+    {
+        ReturnErrorOnFailure(SetICACert(icac));
+    }
+
     if (mOperationalCert == nullptr)
     {
-        mOperationalCert = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(cert.size()));
+        mOperationalCert = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(noc.size()));
     }
     VerifyOrReturnError(mOperationalCert != nullptr, CHIP_ERROR_NO_MEMORY);
-    VerifyOrReturnError(CanCastTo<uint16_t>(cert.size()), CHIP_ERROR_INVALID_ARGUMENT);
-    mOpCertLen          = static_cast<uint16_t>(cert.size());
+    VerifyOrReturnError(CanCastTo<uint16_t>(noc.size()), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This check must be move BEFORE the alloc.

Also, on error of NOC alloc, we need to release the ICA cert.

Strongly suggest the following:

1. Refactor `SetOperationalCert` to `SetOperationalCertsFromCertArray`
2. Make a `SetNOCCert` method that is similar to `SetICACert` in construction
3. Use `SetICACert`/`SetNOCCert` in here rather than having the inlined logic of `SetNOCCert`

Reply:
Thanks, will update.